src/app/browser/qgsinbuiltdataitemproviders.cpp

```diff
@@ -1858,6 +1859,100 @@ void QgsDatabaseItemGuiProvider::populateContextMenu( QgsDataItem *item, QMenu *
         }
       } );
     }
+
+    if ( conn && conn->capabilities().testFlag( QgsAbstractDatabaseProviderConnection::Capability::MoveTableToSchema ) )
+    {
+      QAction *moveToSchemaAction = new QAction( tr( "Move to another Schema…" ), menu );
+      menu->addAction( moveToSchemaAction );
+
+      // this action should sit in the Manage menu. If one does not exist, create it now
+      bool foundExistingManageMenu = false;
+      QList<QAction *> actions = menu->actions();
+      for ( QAction *action : std::as_const( actions ) )
+      {
+        if ( action->text() == tr( "Manage" ) )
+        {
+          action->menu()->addAction( moveToSchemaAction );
+          foundExistingManageMenu = true;
+          break;
+        }
+      }
+      if ( !foundExistingManageMenu )
+      {
+        QMenu *manageLayerMenu = new QMenu( tr( "Manage" ), menu );
+        manageLayerMenu->addAction( moveToSchemaAction );
+        menu->addMenu( manageLayerMenu );
+      }
+
+      const QString connectionUri = conn->uri();
+      const QString providerKey = conn->providerKey();
+
+      connect( moveToSchemaAction, &QAction::triggered, moveToSchemaAction, [item, context, connectionUri, providerKey] {
+        QgsProviderMetadata *md { QgsProviderRegistry::instance()->providerMetadata( providerKey ) };
+        if ( !md )
+          return;
+
+        std::shared_ptr<QgsAbstractDatabaseProviderConnection> conn2( qgis::down_cast<QgsAbstractDatabaseProviderConnection *>( md->createConnection( connectionUri, QVariantMap() ) ) );
+
+        QgsDatabaseSchemaSelectionDialog *dlg = new QgsDatabaseSchemaSelectionDialog( conn2.get() );
+
+        if ( dlg->exec() == QDialog::Accepted )
+        {
+          const QString originalSchemaName = item->parent()->name();
+          const QString tableName = item->name();
+
+          QString errCause;
+          if ( conn2 )
+          {
+            try
+            {
+              QgsTemporaryCursorOverride override( Qt::WaitCursor );
+              conn2->moveTableToAnotherSchema( originalSchemaName, tableName, dlg->selectedSchema() );
```

Comment:
QgsDatabaseSchemaSelectionDialog takes ownership of the connection -- there's no gaurantee `con2` will still exist here (that is, there's a chance that a future refactoring of the QgsDatabaseSchemaSelectionDialog could mean that conn2 won't exist here!).

For safety you should retrieve a new connection for use here.

Reply:
add